mozilla/fx-private-relay
# Create and receive mozmail relay alias

# About this PR
Since we own mozmail.com we want to make sure that Relay can create Relay aliases and get emails relayed with the new domain.

# Acceptance Criteria
- [ ] Add logic to create Relay alias with mozmail.com as the domain
- [ ] When `TEST_MOZMAIL` environment variable is set to `True` users Relay alias is created with mozmail.com domain INSTEAD of relay.firefox.com
- [ ] Even when `TEST_MOZMAIL` is `False` users with `mozilla.com` email address can use mozmail.com
- [ ] Both free and premium users can generate random aliases with mozmail.com
- [ ] Send alias with mozmail domain to the real user's email
- [ ] Increment forward/block on the relay alias

# Related to
[MPP-712](https://mozilla-hub.atlassian.net/browse/MPP-712) and [MPP-714](https://mozilla-hub.atlassian.net/browse/MPP-714)

# Test
## Test that relay address works as is
1. Send email with non-English characters and attachment to the old relay address (relay.firefox.com equivalent)
2. Copy the address by clicking on the copy icon and check that the correct relay address is copied on clipboard
3. Check that email is properly received and the alias on the dashboard (/accounts/profile/) is incremented

## Test that relay address is generated with the mozmail.com when the `TEST_MOZMAIL` environment variable is set to `True`, `MOZMAIL_DOMAIN`, and `RELAY_FIREFOX_DOMAIN` is set
1. Set the environment variable for `TEST_MOZMAIL` to `True` and `MOZMAIL_DOMAIN`, and `RELAY_FIREFOX_DOMAIN` is set to the respective mozmail domain and restart the service
2. Go the the user dashboard (/accounts/profile/)
2. Check that existing Relay Address have the old domain (relay.firefox.com equivalent)
3. Click "Generate New Alias" and check that alias is created with a new domain (mozmail.com equivalent)
4. Copy the address by clicking on the copy icon and check that the correct relay address is copied on clipboard
5. Check that aliases previously created with old domain still has its' old domain and the dash board looks like:
<img width="926" alt="Screen Shot 2021-08-05 at 10 30 58" src="https://user-images.githubusercontent.com/25109943/128379993-a73fad72-3e76-41d5-b7f0-91b47f62eece.png">


## Test that emails are relayed to both alias generated with mozmail.com and existing alias generated with relay.firefox.com
1. Send email to alias generated with old domain (relay.firefox.com equivalent). Send non-English characters and attachments with email size BELOW 150KB.
2. Check that you receive an email and the old domain is reflected properly like: 
<img width="1172" alt="Screen Shot 2021-08-05 at 10 49 12" src="https://user-images.githubusercontent.com/25109943/128381644-059b6388-dfc8-4bcc-b47a-3cea2f7c63f0.png">
3. Send email to alias generated with new domain (mozmail.com equivalent). Send non-English characters and attachments with email size below 150KB.
4. Check that you receive an email and the new domain is reflected properly like: 
<img width="1194" alt="Screen Shot 2021-08-05 at 10 49 20" src="https://user-images.githubusercontent.com/25109943/128381622-b4f21924-f8d0-4837-8847-cdde599c79d3.png">

5. Check that for both aliases, the number of forward/block is incremented
6. Do steps 1-5 again with the alias set to blocked and check that emails are blocked and incremented properly.

## Test email generation on add-on works properly
1. Open the add-on and generate new alias
2. Check that alias generated and filled on the website it was generated on has the mozmail domain
3. Check that on `/accounts/profile/` the alias was generated with the mozmail domain

## Test that users without the mozilla.com email address can use the Relay as is
1. Remove values for the environment variables for `TEST_MOZMAIL`, `MOZMAIL_DOMAIN`, and `RELAY_FIREFOX_DOMAIN`
2. Generate email on dashboard and check that the alias is generated on a relay.firefox.com equivalent domain
3. Send an email with international characters and attachment with size smaller than 150KB
4. Check that email is received in the correct domain (non-mozmail.com)
5. Check that existing aliases also can receive emails with the right address

## Test that users without the mozilla.com email address can use the Relay as is
1. Remove values for the environment variables for `TEST_MOZMAIL`, `MOZMAIL_DOMAIN`, and `RELAY_FIREFOX_DOMAIN`
2. Generate email on dashboard and check that the alias is generated on mozmail.com equivalent domain
3. Send an email with international characters and attachment with size smaller than 150KB
4. Check that email is received in the correct domain (mozmail.com)
5. Check that existing aliases also can receive emails with the right address
